src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -1589,6 +1592,55 @@ def use_radamsa_mutator_plugin(extra_env):
   return True
 
 
+def use_peach_mutator(extra_env, arguments):
+  """Decide whether or not to use peach mutator, and set up all of the
+  environment variables necessary to do so."""
+  # TODO(mpherman): Include architecture info in job definition and exclude
+  #  i386.
+  if environment.platform() != 'LINUX' or environment.get_value(
+      'MEMORY_TOOL') != 'ASAN':
+    return False
+
+  # Extract zip of peach mutator code.
+  peach_dir = os.path.join(environment.get_platform_resources_directory(),
+                           'peach')
+  unzipped = os.path.join(peach_dir, 'mutator')
+  source = os.path.join(peach_dir, 'peach_mutator.zip')
+
+  with ZipFile(source, 'r') as zipped:
+    zipped.extractall(unzipped)
+
+  # Set LD_PRELOAD.
+  peach_path = os.path.join(unzipped, 'peach_mutator', 'src', 'peach.so')
+  extra_env['LD_PRELOAD'] = peach_path
+
+  # Set Python path.
+  # TODO(mpherman): Change this to explicitly point to Python 2
+  new_path = []
+  new_path.append(os.path.join(unzipped, 'peach_mutator', 'src'))
+  new_path.append(
+      os.path.join(unzipped, 'peach_mutator', 'third_party', 'peach'))
+
+  # Append the mutator to the begining of the Python path.
+  new_path = new_path + sys.path
+  extra_env['PYTHONPATH'] = ':'.join(new_path)
+
+  # Select grammar.
+  grammar = fuzzer_utils.extract_argument(arguments, 'grammar')
+  # If the target has no grammar, do not use this strategy.
+  if not grammar:
+    return False
+
+  if not pits.validate(grammar):
```

Comment:
Can we do this check earlier, ideally before extracting?

Reply:
Done